File: dfs/dfs.py
# Using a Python dictionary to act as an adjacency list
graph = {
    'A' : ['B','C'],
    'B' : ['D', 'E', 'B', 'A'],
    'C' : ['F'],
    'D' : ['F'],
    'E' : ['F'],
    'F' : ['G'],
    'G' : ['B'],
}

visited = set() # Set to keep track of visited nodes
def dfs(visited, graph, node, target):
    
    if not graph.get(node):
        return False
    
    if node == target:
        return True
    
    if node not in visited:
        visited.add(node)
        
        for neigbor in graph[node]:
            if dfs(visited, graph, neigbor, target):
                return True
            
    return False
# Driver Code
found = dfs(visited, graph, 'Ds', 'A')

print(found)

File: dfs/dfs.js
const graph = {
  A: ["B", "C"],
  B: ["D", "E"],
  C: ["F"],
  D: [],
  E: ["F"],
  F: [],
};

const visited = new Set();

const dfs = (visited, graph, node, target) => {
  if (node === target) {
    return true;
  }
  visited.add(node);
  let neighbours = graph[node];
  for (let i = 0; i < neighbours.length; i++) {
    if (
      !visited.has(neighbours[i]) &&
      dfs(visited, graph, neighbours[i], target)
    ) {
      return true;
    }
  }
  return false;
};

const isTarget = dfs(visited, graph, "A", "F");

console.log(isTarget ? "yes" : "no");
